### DISCUSSION TITLE
motor.control.limits()

### DISCUSSION THREAD

**Discussion by FLL-Team-24277** [pybricks/support#1715] [General]
Trying to understand control.limits(). This might just be a documentation issue. Or I'm just not understanding....

The documentation states:
Configures the maximum speed, acceleration, and torque.
If no arguments are given, this will return the current values.
The new acceleration and speed limit will become effective when you give a new motor command. Ongoing maneuvers are not affected.

So I was thinking that given the name "control.limits", and the docs saying it sets the maximum speed & acceleration, it would seems that running this command wouldn't set the ACTUAL speed & acceleration, but the MAXIMUM speed & acceleration. In other words, if I ran this

```
rightAttachmentMotor.control.limits(
            acceleration=4000, speed=500
        )
rightAttachmentMotor.run_angle(speed=1000, degrees=360)

```
I thought the motor would run at speed 500, but instead it runs at speed 1000. On the other hand, I set the acceleration (presumably the MAX accel) to 4000 (the default acceleration seems to be 2000 deg/sec^2). But indeed, the motor clearly accelerates much faster.

So, what is the purpose of setting the speed with the control.limits() command? As near as I can tell, it doesn't seem to do anything. On the other hand, it does seem that the control.limit acceleration sets the actual acceleration, rather than the maximum.

**Comment by laurensvalk** [comment]
> I thought the motor would run at speed 500 (...)

Your interpretation of the document is correct.

How are you measuring the speed? If I run the following program, speed limiting does appear to work. The speed peaks at about 500 deg/s, and you can hear that it doesn't get to top speed.

Can you share a program that we can run to reproduce what you are seeing?

![image](https://github.com/user-attachments/assets/aabf8b41-43af-4eef-85e7-4b5593ee16cc)

**Reply by FLL-Team-24277** [reply]
Well, shoot. I had an extra setting of control limits in my base class. Sigh... it does work the way you describe. Silly me.

But I am still confused by the wording that this set the MAXIMUM acceleration. I see it as setting THE acceleration. Are you referring to the obvious point that the acceleration can't be maintained forever, therefore it is the maximum acceleration? Unlike the speed, the acceleration can only be set with the control.limits, so why would would we want to set the max acceleration if there is no other place to try and use something higher? Maybe I am overthinking it...

Anyway, thanks for replying so quickly!!

**Reply by laurensvalk** [reply]
Glad you figured it out!

Yes, the wording for `limits` could be better. This is one of those methods that evolved in terms of functionality over time but the name did not change to preserve compatibility with existing code.

We avoided the word altogether in the blocks as you can see above :)
